soyjs/exec.go

```diff
@@ -25,15 +26,116 @@ type state struct {
 	autoescape   ast.AutoescapeType
 	lastNode     ast.Node
 	options      Options
+	funcsCalled  map[string]string
+	funcsInFile  map[string]bool
+}
+
+type ES6Formatter struct{}
+type ES5Formatter struct{}
+
+func importStyle(s string) string {
+	return strings.Replace(s, ".", "__", -1)
+}
+
+func (f ES6Formatter) Template(name string) (string, string) {
+	return importStyle(name), "export function " + importStyle(name)
+}
+
+func (f ES6Formatter) Call(name string) (string, string) {
+	return importStyle(name), "import { " + importStyle(name) + " } from '" + name + ".js';\n"
```

Comment:
I'd suggest not including the newline here and handling that during the join/write phase